### ISSUE TITLE
deploy on save feature when saving file executing wrong sfdx command. error

### ISSUE THREAD

**Issue by apsoftwaredev** [forcedotcom/salesforcedx-vscode#3482]
### Summary

_Short summary of what is going on or to provide context_.

### Steps To Reproduce:

1.  Save changes to a file in vscode sf project connected to org
2. sends deploy command to terminal.
3.  Terminal reports that sfdx deploy:apex is not a valid SFDX command and gives an error message

›   Warning: deploy:apex is not a sfdx command.
Did you mean help? [y/n]: 
 ›   Error: Run sfdx help for a list of available commands.

### Expected result
deploy to SF

### Actual result
error message in the console.

### Additional information
unable to turn of deploy on save feature.
possibly causing the org to disconnect

![image](https://user-images.githubusercontent.com/1483796/128618125-d8bf1efe-03a1-4bd4-b464-d8627d0778f1.png)
**Salesforce Extension Version in VS Code**:
More Info
Released on
9/21/2017, 09:30:15
Last updated
8/4/2021, 18:01:44
Identifier
salesforce.salesforcedx-vscode-core

**SFDX CLI Version**:
sfdx-cli/7.112.1 wsl-x64 node-v14.17.4
**OS and version**:
Windows 11 insider with WSL2 (vscode running attached to WSL2 Container)

**Comment by xyc** [comment]
Are you using [DX Code Companion extension](https://marketplace.visualstudio.com/items?itemName=MohithShrivastava.dx-code-companion) or https://www.npmjs.com/package/mo-dx-plugin? `deploy:apex` is not a command in or Salesforce extension pack or Salesforce CLI.

**Comment by apsoftwaredev** [comment]
I installed everything according to the installation instructions on Salesforce using npm

The vscode extension was part of the Salesforce extension pack (Expanded). 

More Info
Released on
9/11/2020, 11:18:34
Last updated
8/4/2021, 17:59:19
Identifier
salesforce.salesforcedx-vscode-expanded

What I ended up doing is running the salesforce rc version in a docker container and changing to an insider build of VS Code,  and that doesn't seem to have the bug with the extension.  It might have been fixed already in a newer version. The command was not something I entered.  It was automatically sent to the terminal by the deploy on save feature.   I don't mind the deploy on save feature, but there wasn't a way to disable it in the settings that I saw other than disabling the whole extension.  The docker container version doesn't seem to have that issue.
